aws/src/database/multi_table/update_cautionary_alerts.py

```diff
@@ -0,0 +1,60 @@
+
+from mypy_boto3_dynamodb.service_resource import Table
+
+from aws.src.database.dynamodb.utils.get_dynamodb_table import get_dynamodb_table
+from aws.src.utils.csv_to_dict_list import csv_to_dict_list
+from enums.enums import Stage
+from dateutil import parser
+from datetime import datetime
+
+STAGE = Stage.HOUSING_PRODUCTION
+
+def update_cautionary_alert_in_db_if_tenure_inactive(person_table: Table, alerts_from_csv: list[dict]):
+    """
+    Iterate through the alerts and remove property details if tenure is inactive
+    1. Get person details from dynamo using mmh_id in csv
+    2. check if there is an end date within the tenure object
+    3. if so remove the property details from the csv
+    """
+
+    for alert_item in alerts_from_csv:
+        alert_id = alert_item["id"]
+        mmh_id = alert_item["mmh_id"]
+        if mmh_id is '' or None:
+            alert_item["failed_reason"] = f"mmh_id is null for id {alert_id}"
+            continue
+        person = person_table.get_item(Key={"id": mmh_id}).get("Item")
+
+        person_id = person["id"]
+        if person is None:
+            alert_item["failed_reason"] = f"Could not get person for personId {person_id}"
+            continue
+        tenures = person["tenures"]
+        for tenure_in_person in tenures:
+            end_date = tenure_in_person["endDate"]
+            if end_date is None:
+                continue
+            parsed_end_date: datetime = parser.parse(timestr=end_date.split("T")[0])
+            if parsed_end_date < datetime.now(): 
+                prop_ref = alert_item["property_reference"]
+                if str(tenure_in_person["propertyReference"]) == str(prop_ref):
+                    alert_item["address"] = None
+                    alert_item["property_reference"] = None
+                    alert_item["uprn"] = None    
+                    alert_item["door_number"] = None          
```

Comment:
This probably already was considered, but I'll ask just to make sure:

The Cautionary Alert is associated to 2 things: the property, and the person.

If the tenure is inactive, this data update for which this tsv spreadsheet is prepared will remove the property relation from the Cautionary Alert. However, that alert is still related the person.

I imagine there are some cases, where that person may have moved to another Council managed property.  In such an event:
- Wouldn't we want to find out what the property is & update the Cautionary Alert to point to this new property?

Reply:
this script was just for a one-off scenario that has already taken place, we were not told by Liz about any people who have moved to another property. The Cautionary Alert is still attached to the person. In a worst-case scenario, we would probably get a Halo ticket to say the Alert is not attached to the property. If we get lots of Halo tickets then we can look into creating a script for it.